downstairs/src/lib.rs

```diff
@@ -1034,13 +1114,15 @@ where
                             new_upstairs_connection, upstairs_connection
                         );
 
-                        if let Err(e) = fw.send(Message::YouAreNoLongerActive {
-                            new_upstairs_id:
-                                new_upstairs_connection.upstairs_id,
-                            new_session_id:
-                                new_upstairs_connection.session_id,
-                            new_gen: new_upstairs_connection.gen,
-                        }).await {
+                        if let Err(e) = reply_channel_tx
```

Comment:
I think I agree with the overall vibes: it's possible to get into some kinda weird situations, but nothing actually dangerous.  We also have the upstairs / session ID checks, which would prevent any actual funny business.

For what it's worth, this will also get simpler later on, because `proc_task` itself is removed. 

(One possible cleanup – for later – would be to intentional cancellation only stop `recv_task`, then have everything else notice the cancellation when their `mpsc::Receiver::recv` runs dry)

Reply:
> One possible cleanup – for later – would be to intentional cancellation only stop recv_task, then have everything else notice the cancellation when their mpsc::Receiver::recv runs dry

That sounds good, fwiw.